### ISSUE TITLE
Zero-cost assets not included in unrealized gains

### ISSUE THREAD

**Issue by blais** [beancount/beancount#42]
**[Original report](https://bitbucket.org/blais/beancount/issue/42) by clemente (Bitbucket: [clemente](https://bitbucket.org/clemente), GitHub: [clemente](https://github.com/clemente)).**

----------------------------------------

If I get new units gifted, they aren't included in the unrealized gains.

~~~~
plugin "beancount.plugins.unrealized"

1970-01-01 open Assets:Stocks:NYA
1970-01-01 open Expenses:Financial:Commissions
1970-01-01 open Assets:Current
1970-01-01 open Earnings:Dividends:NYA

2010-01-10 ! "Buy"
  Assets:Stocks:NYA 20 NYA {15 EUR}
  Expenses:Financial:Commissions 10 EUR
  Assets:Current -310 EUR

2012-07-02 ! "I received 1 new share in dividend, without paying"
  Assets:Stocks:NYA 1 NYA {0 EUR}
  Earnings:Dividends:NYA -0 EUR

2014-11-13 balance Assets:Stocks:NYA 21 NYA
2014-11-13 price NYA 16 EUR ; I should have (* 16 21) = 336 EUR
~~~~

I saw:

~~~~
Unrealized gain for 20 units of NYA (price: 16.0000 EUR as of 2014-11-13, average cost: 15.0000 EUR)
~~~~

Instead of „… for 21 units“. Average cost should alse have gone down from 15.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Fixed issue #42, minor bug with the calculation of holdings for zero-cost positions.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Thank you Daniel!
